src/promptflow-devkit/promptflow/batch/_batch_engine.py

```diff
@@ -288,19 +290,37 @@ def _copy_previous_run_result(
         """
         try:
             previous_run_results = []
-            aggregation_nodes = {node.name for node in self._flow.nodes if node.aggregation}
-            for i in range(len(batch_inputs)):
+            if not self._is_eager_flow:
+                aggregation_nodes = {node.name for node in self._flow.nodes if node.aggregation}
+            for i, _ in enumerate(batch_inputs):
                 previous_run_info: FlowRunInfo = resume_from_run_storage.load_flow_run_info(i)
 
-                if previous_run_info and previous_run_info.status == Status.Completed:
-                    # UI uses root_run_id  to link the base path in datastore with the run_info of line.
-                    # Thus the root_run_id needs to be the current batch run id.
-                    previous_run_info.root_run_id = run_id
-                    previous_run_info.parent_run_id = run_id
+                if not previous_run_info or previous_run_info.status != Status.Completed:
+                    continue
 
-                    # Load previous node run info
-                    previous_node_run_infos = resume_from_run_storage.load_node_run_info_for_line(i)
+                # UI uses root_run_id  to link the base path in datastore with the run_info of line.
+                # Thus the root_run_id needs to be the current batch run id.
+                previous_run_info.root_run_id = run_id
+                previous_run_info.parent_run_id = run_id
+
+                # Deepcopy to avoid modifying the original object when serializing image
+                self._storage.persist_flow_run(previous_run_info)
+                previous_run_output = deepcopy(previous_run_info.output)
+                previous_run_output_in_line_result = self._multimedia_processor.persist_multimedia_data(
+                    previous_run_output, output_dir
+                )
 
+                if self._is_eager_flow:
+                    # Directly create LineResult object for previous line result
+                    previous_line_result = LineResult(
+                        output=previous_run_output_in_line_result,
+                        aggregation_inputs=previous_run_output_in_line_result,
```

Comment:
Just curious, does flex flow has `aggregation_inputs`?

Reply:
After syncing with Lina, we confirmed that flex flow can support aggregation and its aggregation input only comes from results of every line. For example the input of __aggregate__ comes from the results of __call__. Different from dag flow run, it doesn't support aggregation inputs from flow input.
<img width="584" alt="1713496974651" src="https://github.com/microsoft/promptflow/assets/39176492/8bb6e89c-748a-4332-9536-1bc66dec2c81">

Reply:
Please note user can get values in init in __aggretate__